llvm/lib/Target/AArch64/AArch64InstrFormats.td

```diff
@@ -6055,6 +6055,15 @@ multiclass SIMDThreeSameVectorFML<bit U, bit b13, bits<3> size, string asm,
                                          v4f32, v8f16, OpNode>;
 }
 
+multiclass SIMDThreeSameVectorMLA<bit Q, string asm>{
```

Comment:
What's the reason for using multiclasses if they contain a single record each?

Reply:
Maybe you are correct.
But in my view, it's more clear to use a multiclass in this case because the class is called Dot while the multiclass instance is for MLA, so it's more clear.
What do you think ?

Reply:
We could use the class  BaseSIMDThreeSameVectorDot when defining , but she is  following the pattern of the other instructions(despite the fact they have more than one definition).
Also we can hide the size of the instructions in the multiclass.

Reply:
An alternative would be to define a new class which inherits from `BaseSIMDThreeSameVectorDot` and handles the common values for all the definitions. For example:
```
class SIMDThreeSameVectorMLA<bit Q, string asm>
    : BaseSIMDThreeSameVectorDot<Q, 0b0, 0b11, 0b1111, asm, ".8h", ".16b",
                                                              V128, v8f16, v16i8, null_frag>;
```
Using multiclasses for consistency sounds reasonable, though, if you prefer to keep the current implementation.